src/CodeStyle/Core/CodeFixes/AbstractFormattingAnalyzer.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    /// <summary>
+    /// This analyzer implements <see cref="DiagnosticAnalyzer"/>, but wraps the true formatting analyzer
+    /// implementations. Since the compiler doesn't provide the use of workspace-layer APIs for analyzers, we need to
+    /// provide our own copy of the workspace assemblies and load them using reflection if they are not already loaded
+    /// in the current process.
```

Comment:
i'm not getting the reason for reflection, as opposed to just a normal reference...